src/main/java/com/pgs/booking/service/UserService.java

```diff
@@ -13,15 +19,53 @@
 import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationToken;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
 import java.util.Optional;
 
 @Service
 @RequiredArgsConstructor
 public class UserService implements UserDetailsService, AuthenticationUserDetailsService<PreAuthenticatedAuthenticationToken> {
 
     private final UserRepository userRepository;
+    private final UserDtoMapper userDtoMapper;
+    private final CreateUserDtoMapper createUserDtoMapper;
+    private final RoleRepository roleRepository;
     private final TokenStore tokenStore;
 
+    public UserDto getSingleUser(Long id) {
+        var user = getUserById(id);
+        return userDtoMapper.mapToUserDto(user);
+    }
+
+    public UserDto addUser(CreateUserDto createUserDto) {
+        var savedUser = userRepository.save(createUserDtoMapper.mapToUser(createUserDto));
```

Comment:
in the request we provide password as raw value, but in DB we want it as hash. I don't see you encrypting it anywhere.